src/abstractblockarray.jl

```diff
@@ -99,15 +100,24 @@ end
     blockcheckbounds_indices(Bool, blockaxes(A), i)
 end
 
+checkbounds(A::AbstractArray{T, N}, i::Block{N}) where {T,N} = blockcheckbounds(A, i)
```

Comment:
IIUC
```suggestion
checkbounds(::Type{Bool}, A::AbstractArray{T, N}, i::Block{N}) where {T,N} = blockcheckbounds(A, i)
```
and similarly elsewhere

Reply:
This is deliberate in order to implement #458, otherwise they will call the Base version: https://github.com/JuliaLang/julia/blob/309b1b158f59485772d5f5fe0a762f20185cf799/base/abstractarray.jl#L695-L699 and will throw a `BoundsError` instead of a `BlockBoundsError`.

Reply:
Oh, I see. Perhaps `Base` should make the error-throwing function public so that a method may be added here. In the meanwhile, this approach is fine.